xcm/src/v0/traits.rs

```diff
@@ -121,21 +122,38 @@ impl Outcome {
 }
 
 pub trait ExecuteXcm<Call> {
+	type Call;
 	fn execute_xcm(origin: MultiLocation, message: Xcm<Call>, weight_limit: Weight) -> Outcome;
 }
 
 impl<C> ExecuteXcm<C> for () {
+	type Call = C;
 	fn execute_xcm(_origin: MultiLocation, _message: Xcm<C>, _weight_limit: Weight) -> Outcome {
 		Outcome::Error(Error::Unimplemented)
 	}
 }
 
+/// Utility for sending an XCM message.
+///
+/// These can be amalgamted in tuples to form sophisticated routing systems.
 pub trait SendXcm {
```

Comment:
I was reading this few hours ago and this was a fundamental part of the code without docs, thanks for filling 👍